File: tsp-solver-frontend/src/app/graph-spp/Edge.ts
import {RoadType} from "./RoadType";

export class Edge {
  id: string;
  source: string;
  target: string;
  distance: number;
  roadType: RoadType;
  bearingCapacity: number
  color: string

  constructor(id: string, source: string, target: string, distance: number, roadType: RoadType,
              bearingCapacity: number, color: string) {
    this.id = id;
    this.source = source;
    this.target = target;
    this.distance = distance;
    this.roadType = roadType;
    this.bearingCapacity = bearingCapacity;
    this.color = color;
  }

}

File: tsp-solver-backend/src/main/java/pl/edu/pbs/spp/Edge.java
package pl.edu.pbs.spp;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Edge {

    private String id;
    private String source;
    private String target;
    private int roadTypeId;
    private double distance;
    private double bearingCapacity;

    /**
     * {@link Edge#cost} = {@link Edge#distance} * {@link RoadType#getWeight()}
     */
    private double cost;
}
